src/StructTree/StructTree.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { getAttributes } from './utils';
+import type { StructTreeProps } from './types';
+import type { StructTreeNode } from 'pdfjs-dist/types/src/display/api';
+
+export default function StructTree({ node }: StructTreeProps) {
+  const attributes = useMemo(() => getAttributes(node), [node]);
+
+  const childNodes = useMemo(() => {
+    if (
+      node.children &&
+      !(node.children.length === 1 && node.children[0] && 'id' in node.children[0])
```

Comment:
I don't understand what this line does

Reply:
The structure that gets generated by the pdfjs function for struct tree is a bit weird. You pretty much want to stop going deeper once you encounter a node with 1 element which has an id. Otherwise you want to keep going